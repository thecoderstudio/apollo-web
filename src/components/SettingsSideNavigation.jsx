import React from 'react';
import styled from 'styled-components';
import { matchPath } from 'react-router';
import Link from './Link';
import Icon from './Icon';
import media from '../util/media';

const Navigation = styled.div`
  max-width: 225px;
  width: 100%;
  background-color: ${props => props.theme.lightBlack};
  box-shadow: inset ${props => props.theme.boxShadow};
  padding-top: 32px;

  ${media.phone`
    width: 45px;
  `}

  transition: all 1s ease;
  overflow: hidden;
`;

const StyledIcon = styled(Icon)`
  margin: 0 8px;

  ${media.phone`
    margin: 0 auto;
  `}
`;

const StyledLink = styled(({ active, ...props }) => <Link {...props} />)`
  padding: 12px;
  margin-tpropsx;
  border-radius: 4px;
  width: 100%;
  float: left;
  overflow: hidden;
  text-overflow: ellipsis;
  box-spropsborder-box;
  white-space: nowrap;

  background-color: ${props => props.active ? props.theme.selectedBlack : props.theme.lightBlack};

  &:hover {
    background-color: ${props => props.theme.selectedBlack};
  }
`;

const TextWrapper = styled.span`
  margin: 0 8px;

  ${media.phone`
    display: none;
  `}
`;

class SettingsSideNavigation extends React.PureComponent {
  constructor(props) {
    super(props);
    this.checkIfPathIsActive = this.checkIfPathIsActive.bind(this);
  }

  checkIfPathIsActive(path) {
    const { location } = this.props;
    const match = matchPath(location.pathname, {
      path,
      exact: true,
      strict: false
    });
    if (!match) {
      return false;
    }
    return match.isExact;
  }

  render() {
    const { className } = this.props;
    return (
      <Navigation className={className}>
        <StyledLink
          active={this.checkIfPathIsActive('/settings/user_settings')}
          to="/settings/user_settings"
        >
          <StyledIcon className="fas fa-user-cog" />
          <TextWrapper>User settings</TextWrapper>
        </StyledLink>
      </Navigation>
    );
  }
}

export default SettingsSideNavigation;
