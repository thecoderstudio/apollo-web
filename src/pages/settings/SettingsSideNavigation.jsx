import React from 'react';
import styled from 'styled-components';

const Navigation = styled.div`
  max-width: 300px;
  width: 100%;
  background-color: ${(props) => props.theme.lightBlack};
`;

function SettingsSideNavigation(props) {
  return (
    <Navigation className={props.className}>
      hahahahha
    </Navigation>
  );
}

export default SettingsSideNavigation;
