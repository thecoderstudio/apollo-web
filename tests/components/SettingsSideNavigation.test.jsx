import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SettingsSideNavigation from '../../src/components/SettingsSideNavigation';

function getComponent(props) {
  return renderer.create(
    <BrowserRouter>
      <SettingsSideNavigation {...props} />
    </BrowserRouter>
  );
}

describe('Settings side navigation', () => {
  let props;

  beforeEach(() => {
    props = {
      location: { pathname: '/path' }
    };
  });

  it('renders correctly', () => {
    let tree = getComponent(props).toJSON();
    expect(tree).toMatchSnapshot();

    tree = getComponent({
      location: { pathname: '/settings/user_settings' }
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('correctly check if path matches', () => {
    const tree = getComponent(props);
    const instance = tree.root;
    const nav = instance.findByType(SettingsSideNavigation);

    expect(nav.instance.checkIfPathIsActive('/path')).toBe(true);
    expect(nav.instance.checkIfPathIsActive('/something_else')).toBe(false);
  });
});
