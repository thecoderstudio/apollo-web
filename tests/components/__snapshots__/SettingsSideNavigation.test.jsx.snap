// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings side navigation renders correctly 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  opacity: 0.90;
}

.c3 {
  cursor: pointer;
}

.c4 {
  margin: 0 8px;
}

.c2 {
  padding: 12px;
  margin-top: 24px;
  border-radius: 4px;
  width: 100%;
  float: left;
  overflow: hidden;
  text-overflow: ellipsis;
  box-sizing: border-box;
  white-space: nowrap;
}

.c5 {
  margin: 0 8px;
}

.c0 {
  max-width: 225px;
  width: 100%;
  box-shadow: inset;
  padding-top: 32px;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  overflow: hidden;
}

@media (max-width:47.9375em) {
  .c4 {
    margin: 0 auto;
  }
}

@media (max-width:47.9375em) {
  .c5 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    width: 45px;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="/settings/user_settings"
    onClick={[Function]}
  >
    <i
      className="c3 c4 fas fa-user-cog"
    />
    <span
      className="c5"
    >
      User settings
    </span>
  </a>
</div>
`;

exports[`Settings side navigation renders correctly 2`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  opacity: 0.90;
}

.c3 {
  cursor: pointer;
}

.c4 {
  margin: 0 8px;
}

.c2 {
  padding: 12px;
  margin-top: 24px;
  border-radius: 4px;
  width: 100%;
  float: left;
  overflow: hidden;
  text-overflow: ellipsis;
  box-sizing: border-box;
  white-space: nowrap;
}

.c5 {
  margin: 0 8px;
}

.c0 {
  max-width: 225px;
  width: 100%;
  box-shadow: inset;
  padding-top: 32px;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  overflow: hidden;
}

@media (max-width:47.9375em) {
  .c4 {
    margin: 0 auto;
  }
}

@media (max-width:47.9375em) {
  .c5 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    width: 45px;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="/settings/user_settings"
    onClick={[Function]}
  >
    <i
      className="c3 c4 fas fa-user-cog"
    />
    <span
      className="c5"
    >
      User settings
    </span>
  </a>
</div>
`;
